typing/typedecl.ml

```diff
@@ -1121,8 +1124,14 @@ let transl_type_decl env rec_flag sdecl_list =
     check_well_founded_decl new_env (List.assoc id id_loc_list) (Path.Pident id)
       decl to_check)
     decls;
+  (* Add abstract declarations to the environment to avoid strange error
+     messages (#12334) *)
+  let abs_env =
+    List.fold_left2 (enter_type ~abstract_abbrevs:false rec_flag)
+      env sdecl_list ids_list in
```

Comment:
I would have expected `abstract_abbrevs:true` here. Are you using `true` anywhere?

Reply:
Creating a whole new env here seems potentially expensive. Is there instead a way to catch and reinterpret the error message in `check_regularity`?

Reply:
@gasche Well-spotted. I did various testing, and using true or false doesn't seem to have much impact, but it should be `true` here for keeping the original behavior. Fixed now

@goldfirere Actually, this is not very expensive. We are just building a dummy env here, which contains only abstract types, and whose type parameters are not constrained. Using the original environment was really a dirty hack.